pkg/agent/client/client.go

```diff
@@ -311,7 +311,7 @@ func (c *client) fetchEntries(ctx context.Context) ([]*types.Entry, error) {
 	if err != nil {
 		c.release(connection)
 		c.c.Log.WithError(err).Error("Failed to fetch authorized entries")
-		return nil, fmt.Errorf("failed to fetch authorized entries: %v", err)
+		return nil, err
```

Comment:
Can keep this "embellishment" if we wrap the error and then `isShutdownError` changes to unwrap?
```suggestion
		return nil, fmt.Errorf("failed to fetch authorized entries: %w", err)
```

See https://golang.org/pkg/errors/#Unwrap

It would probably be good to wrap all of the errors (via `%w`) on all of the errors coming out of the client so we can react to status messages coming out of other authenticated RPCs (i.e. BatchNewX509SVID)

Reply:
I also wrapped `BatchNewX509SVID` in the last commit. What do you think about `NewJWTSVID`, `RenewAgent` and `GetFederatedBundle`? The first one it's not being wrapped, while the other two are not being embellished.

Reply:
As long as each error is either wrapped or not embellished, we are ok. If you want to update those other methods to wrap and embellish the error message, I'd be fine with that.